WORKSPACE

```diff
@@ -0,0 +1,25 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+#################################
+#### Bazelized external libs ####
+local_repository( name = "xbyak" , path = "src/camlsnark_c/libsnark-caml/depends/xbyak" )
+# http_archive(
+#     name = "xbyak",
+#     urls = ["https://github.com/obazl/xbyak/archive/bzl-1.0.tar.gz"],
```

Comment:
Similar here: I think when it comes time for us to switch over, I would personally prefer that this be a submodule with a `path`.

Reply:
We can have both. In this case, `src/camlsnark_c/libsnark-caml/depends/xbyax` can be a submodule, or not, Bazel doesn't care.  So the migration path is probably: install Bazel support, with repos pointing to the embedded code. Then switch the embedded code to submodules; this will have no effect on Bazel.  Then, if we want to move to pure bazel, delete the submodules and depend only on the Bazel repos.